rest/data/interface.go

```diff
@@ -77,6 +77,13 @@ type Connector interface {
 	// RestartBuild is a method to restart the build matching the same BuildId.
 	RestartBuild(string, string) error
 
+	// Find project variables matching given projectId.
+	FindProjectVarsById(string) (*restModel.APIProjectVars, error)
+	// UpdateProjectVars updates the project using the variables given in the model,
+	// and additionally accepts a list of variables to remove.
+	// If successful, updates the given projectVars with the updated projectVars.
+	UpdateProjectVars(*model.ProjectVars, []string) error
```

Comment:
I think that we should an update method and a separate delete vars method.

Reply:
I'm pushing a refactor soon which uses the API project var struct instead, which includes varsToDelete already, so I think this change isn't necessary.

Reply:
FindAndModify still accepts varsToDelete as a variable, but that's more hidden so I would argue for it rather than making two database calls.